lib/events/keyverificationevent.cpp

```diff
@@ -0,0 +1,193 @@
+// SPDX-FileCopyrightText: 2021 Carl Schwan <carlschwan@kde.org>
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "keyverificationevent.h"
+
+using namespace Quotient;
+
+KeyVerificationRequestEvent::KeyVerificationRequestEvent(const QJsonObject &obj)
+    : Event(typeId(), obj)
+{}
+
+QString KeyVerificationRequestEvent::fromDevice() const
```

Comment:
Ok - there's a lot of boilerplate here. Partially it can be reduce with the magic `Event::contentPart<>` method that is, effectively, `fromJson<ContentType>(contentJson()[KeyId])`. For simpler things like `QString` it doesn't save much but when you get to `QStringList` it really saves quite a few characters. I recommend using it across the board because it's conveys the intent much clearer.

Further savings would involve devising a macro (or rather macros) that would declare/define content accessors, along the lines of:
```cplusplus
#define CONTENT_GETTER_IMPL(PartType_, PartName_)                 \
    {                                                             \
        static const auto JsonKey = toSnakeCase(#PartName_##_ls); \
        return contentPart<PartType_>(JsonKey);                   \
    }

#define CONTENT_GETTER(PartType_, PartName_) \
    PartType_ PartName_() const CONTENT_GETTER_IMPL(PartType_, PartName_)
```
(the above produces an inline getter but a similar `DEFINE_CONTENT_GETTER` can be done for out-of-line definitions, reusing `CONTENT_GETTER_IMPL`). I think of introducing those macros in `event.h`, given how pervasive the pattern is across libQuotient. If you don't want macros, feel free to keep proper method definitions but do use `contentPart()`, please.

Reply:
I've done the contentpart stuff for now, i'll do the rest once the macro lands

Reply:
After playing around with the concept a bit, I think I'll only introduce the inline version (pretty much what I wrote above), as out-of-line `DECLARE_*` and `DEFINE_*` macros don't save that much and ultimately are less readable than properly spelled declarations/definitions - meaning that this code will be accepted as it is now. I might eventually (when we move to modules in 5 years?) inline all such things but that's a very different story.